src/js/controllers/ctrl-media-import.js

```diff
@@ -134,10 +145,18 @@ app.controller('MediaImportCtrl', function ($scope, $window, $timeout) {
 					MIME_MAP[type].forEach((subtype) => {
 						extractedTypes = [...extractedTypes, subtype.split('/')[1]]
 					})
-
 					allowedFileExtensions = [...allowedFileExtensions, ...extractedTypes]
 				}
-			})
+			});
+
+			// obj file extensions MIME_TYPE changes based on upload or download.
+			// From user to server the MIME_TYPE is application/octet-stream.
+			// From server to user the MIME_TYPE is text/plain.
+			// Statement allows for the seeing of obj files in the list of previously uploaded models.
+			if (allowedFileExtensions.indexOf('octet-stream') > -1) {
+				allowedFileExtensions.push('obj');
+				allowedFileExtensions.shift();
```

Comment:
Unclear why `.shift()` is being used here - if you're trying to remove a specific item from the array, find that item's index first and only remove that item. Assuming the thing you're getting rid of will always be the first element can be dangerous. A comment would be helpful either way.